test/gtest/ucp/ucp_test.cc

```diff
@@ -987,7 +988,10 @@ void ucp_test_base::entity::ep_destructor(ucp_ep_h ep, entity *e)
     ucs_status_t        status;
     ucp_tag_recv_info_t info;
     do {
-        e->progress();
+        const ucp_test *test = dynamic_cast<const ucp_test*>(e->m_test);
```

Comment:
is it to progress TCP flush on the remote side?

Reply:
yes, it was the idea. otherwise, the test hangs.